src/dragonball/src/tracer.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) 2023 Alibaba Cloud
+// Copyright (c) 2023 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+use std::cmp::min;
+use std::fmt::{Debug, Formatter};
+use std::sync::{Arc, Mutex};
+
+use anyhow::Result;
+use log::{info, warn};
+use opentelemetry::global;
+use tokio::runtime::Runtime;
+use tracing::subscriber::DefaultGuard;
+use tracing::{span::EnteredSpan, subscriber::NoSubscriber, trace_span, Subscriber};
+use tracing_opentelemetry::OpenTelemetryLayer;
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::Registry;
+
+const DEFAULT_JAEGER_URL: &str = "http://localhost:14268/api/traces";
+
+/// dragonball tracer error.
+#[derive(Debug, thiserror::Error)]
+pub enum DragonballTraceError {
+    /// verify jaeger config failed.
+    #[error("config verify failed: {0}")]
+    ConfigVerifyFailed(anyhow::Error),
```

Comment:
add #source to expand the error stack, and we shouldn't use anyhow in dragonball

Reply:
Fixed.